### PR TITLE
refactor: start options

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR is to refactor configuration structs of datanode/frontend/standalone, allowing datanode/frontend to accept a list of metasrv addresses. Also this PR removes unwrapping `FrontendOptions::metasrv_addr`, instead, a user-friendly error with stacktrace will be raised.

This PR has been tested in following scenarios:
- start in standalone mode
    - with all options left to default 
    - with config file in `config` directory
- start datanode/frontend
    - with all options left to default 
    - with config file  in `config` directory
    - specify metasrv addr in cmd arguments, other options left to default

## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: config/datanode.example.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ type = 'File'
 data_dir = '/tmp/greptimedb/data/'
 
 [meta_client_opts]
-metasrv_addr = '1.1.1.1:3002'
+metasrv_addr = ['1.1.1.1:3002']
```
Comment (evenyag) [inline config/datanode.example.toml:14]: Use `addrs`?
```suggestion
metasrv_addrs = ['1.1.1.1:3002']
```

File: src/cmd/src/standalone.rs [hunk 2]
```diff
@@ -105,7 +106,7 @@ impl StandaloneOptions {
             prometheus_options: self.prometheus_options,
             mode: self.mode,
             datanode_rpc_addr: "127.0.0.1:3001".to_string(),
-            metasrv_addr: None,
+            meta_client_opts: Some(MetaClientOpts::default()),
```
Comment (waynexia) [inline src/cmd/src/standalone.rs:109]: Is meta required in standalone mode?

Comment (v0y4g3r) [reply src/cmd/src/standalone.rs:109]: > Is meta required in standalone mode?

No

Comment (waynexia) [reply src/cmd/src/standalone.rs:109]: >No

Then how about default this field to `None`

**Review by waynexia — APPROVED** [review]
👍

File: config/datanode.example.toml [hunk 3]
```diff
@@ -11,7 +11,7 @@ type = 'File'
 data_dir = '/tmp/greptimedb/data/'
 
 [meta_client_opts]
-metasrv_addr = '1.1.1.1:3002'
+metasrv_addrs = ['1.1.1.1:3002']
```
Comment (killme2008) [inline config/datanode.example.toml:14]: Change to `127.0.0.1:3002`? It's a mistake here.

File: config/frontend.example.toml [hunk 4]
```diff
@@ -3,7 +3,7 @@ datanode_rpc_addr = '127.0.0.1:3001'
 http_addr = '127.0.0.1:4000'
 
 [meta_client_opts]
-metasrv_addr = '1.1.1.1:3002'
+metasrv_addrs = ['1.1.1.1:3002']
```
Comment (killme2008) [inline config/frontend.example.toml:6]: The same with above.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**